Sources/TootSDK/Models/ScheduledPostParams.swift

```diff
@@ -78,4 +78,21 @@ public struct ScheduledPostParams: Codable, Sendable {
         case contentType = "content_type"
         case inReplyToConversationId = "in_reply_to_conversation_id"
     }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.text = try? container.decode(String.self, forKey: .text)
+        self.mediaIds = try? container.decode([String].self, forKey: .mediaIds)
+        self.poll = try? container.decode(CreatePoll.self, forKey: .poll)
+        self.inReplyToId = try? container.decode(String.self, forKey: .inReplyToId)
+        // Mastodon incorrectly returns this as string
+        self.sensitive = try? container.decodeBoolFromString(forKey: .sensitive)
+        self.spoilerText = try? container.decode(String.self, forKey: .spoilerText)
```

Comment:
For all optional fields, we should use `decodeIfPresent` as they may not be available in the payload.

Reply:
Here it is from my console (also reported on https://github.com/mastodon/mastodon/issues/24645 )
```json
[
{
"id" : "1023",
"scheduled_at" : "2023-08-31T08:12:00.000Z",
"params" : {
"scheduled_at" : null,
"in_reply_to_id" : null,
"media_ids" : null,
"application_id" : 91595,
"poll" : null,
"spoiler_text" : "Warn",
"idempotency" : null,
"allowed_mentions" : null,
"text" : "Test",
"language" : "en",
"visibility" : "direct",
"with_rate_limit" : false,
"sensitive" : "true"
},
"media_attachments" : [

]

}
]
```

Reply:
[
{
"id" : "1023",
"scheduled_at" : "2023-08-31T08:12:00.000Z",
"params" : {
"scheduled_at" : null,
"in_reply_to_id" : null,
"media_ids" : null,
"application_id" : 91595,
"poll" : null,
"spoiler_text" : "Warn",
"idempotency" : null,
"allowed_mentions" : null,
"text" : "Test",
"language" : "en",
"visibility" : "direct",
"with_rate_limit" : false,
"sensitive" : "true"
},
"media_attachments" : [

]

}
]

Reply:
Added decodeIfPresent to ScheduledPostParams and also for one property in FeaturedTag

Reply:
added scheduled_status_sensitive.json to the test resources